source/templates.tex

```diff
@@ -2800,24 +2800,26 @@
 \end{example}
 
 \pnum
-The instantiation of a pack expansion
-that is neither a \tcode{sizeof...} expression
-nor a \grammarterm{fold-expression}
-produces a
-list of elements
-$\mathtt{E}_1,$ $\mathtt{E}_2,$ $\cdots,$ $\mathtt{E}_N$,
+The instantiation of a pack expansion considers
+items $\tcode{E}_1, \tcode{E}_2, \dotsc, \tcode{E}_N$,
 where
-$N$ is the number of elements in the pack expansion parameters. Each
-$\mathtt{E}_i$ is generated by instantiating the pattern and
+$N$ is the number of elements in the pack expansion parameters.
+Each $\mathtt{E}_i$ is generated by instantiating the pattern and
 replacing each pack expansion parameter with its $i^\text{th}$ element.
 Such an element, in the context of the instantiation, is interpreted as
 follows:
 \begin{itemize}
 \item
-if the pack is a template parameter pack, the element is a template
-parameter\iref{temp.param} of the corresponding kind (type or
-non-type) designating the $i^\text{th}$
-corresponding type or value template argument;
+if the pack is a template parameter pack, the element is
+an \grammarterm{id-expression}
+(for a non-type template parameter pack),
+a \grammarterm{typedef-name}
+(for a type template parameter pack declared without \tcode{template}), or
+a \grammarterm{template-name}
+(for a type template parameter pack declared with \tcode{template})
+designating the $i^\text{th}$ template parameter
+that resulted from instantiation of
+the template parameter pack.
```

Comment:
"that resulted from instantiation of the template parameter pack." seems like it's only suitable for the template parameter pack itself is a pack expansion(it might be non-type template pack or type template pack declared with template). So, Is it necessary to add a restriction to that sentence, that is "if the template parameter pack itself is a pack expansion"?

Reply:
I'm not seeing your point. Could you given an example, please?

Reply:
My point is that "instantiation of the template parameter pack" sounds like the template parameter pack is a pack expansion, such a description has the same manner as that of the function template parameter(which is itself a pack expansion). `template<typename...T> struct A{ template<T...id> struct B{};};`. In this example, the template parameter pack of the nested class template is a pack expansion, hence, when using the pack `id` in some context, we would say that replace `id` to designate ith template parameter **that resulted from instantiation of the template parameter pack**. However,  a type template parameter pack declared without `template` is never a pack expansion. Furthermore, such a `template<int...N> struct C{T b = {N...};};` where `int...C` is not a pack expansion and when instantiating the pattern `N`, each element merely designates the ith template parameter; I mean the result is not instantiated from something, instead, it just designates the corresponding template parameter according to the template arguments.

Reply:
@jensmaurer: any final thoughts on this?